src/signalcorr.jl

```diff
@@ -61,34 +60,31 @@ where each column in the result will correspond to a column in `x`.
 
 The output is not normalized. See [`autocor!`](@ref) for a method with normalization.
 """
-function autocov!(r::RealVector, x::RealVector, lags::IntegerVector; demean::Bool=true)
+function autocov!(
+    r::AbstractVector,
+    x::AbstractVector,
+    lags::AbstractVector{<:Integer},
+    z=zeros(typeof(zero(eltype(x)) / 1), length(x));
```

Comment:
What's the reason for `z` here? I assume it is intended to fix the type instability? I think this should be done differently: The implementation here will lead to different containers (always a `Vector`) than the current implementation.

Moreover, it is not used in the non-`demean` case below, and in general does not have to be filled with zeros.